### PR TITLE
Update AtScale Perf workbook for context-free mode

### PR DESCRIPTION
**Pull request by andrewki-msft**
Update Performance Analysis workbook to work in context-free mode (via Azure Monitor Workbooks ToC) by selecting a random workspace if there is no workspace passed in (context-free mode). Update includes handling cases where a workspace may not have any onboarded machines to Service Map, so a disclaimer has been added to direct the user how to onboard VMs. The title has been removed in-context mode since the workbook title is now displayed in the upper left-hand corner.

Context-free mode
image

Context-free mode (needs onboarding)
image

In-context mode (traditional)
image

### CODE REVIEW

**Comment by acearun** [comment]
Your changes will place the Performance template in a gallery group called 'Performance Analysis' based on the categoryResources.json file. You want the group to be Virtual Machines in Azure Monitor (and order 100 - at the top). Use the category resource override at the gallery level to give it a different group string reference. Eric can show you how to do it..

**Comment by andrewki-msft** [comment]
@ericc1103 I am seeing duplicate entries in the workbook galleries after following the README

Azure Monitor workbooks gallery
image

AtScale VM Insights workbooks gallery
image

**Comment by acearun** [comment]
@ericc1103, can you investigate the duplicate template issue Andrew mentions?

**Comment by ericc1103** [comment]
> @ericc1103 I am seeing duplicate entries in the workbook galleries after following the README. I see this behavior both in PROD and PPE.
> 
> Azure Monitor workbooks gallery
> image
> 
> AtScale VM Insights workbooks gallery
> image

@andrewki-msft Are you filtering a gallery by resourceType with `microsoft.operationalinsights/workspaces`?

**Comment by andrewki-msft** [comment]
Working on a fix when loading the workbook in context-free mode, some defaults aren't being pre-selected

**Comment by andrewki-msft** [comment]
> Working on a fix when loading the workbook in context-free mode, some defaults aren't being pre-selected

Okay so it looks like this has been fixed where a default subscription wasn't being selected, I copied some low-level configuration from a different workbook which worked. I will have to dig deeper into this as a separate task but at least this workbook looks good to go as soon as the virtual category has been fixed.

**Comment by andrewki-msft** [comment]
> Why do we need Onboarding message? Can you please remove 'This Workspace requires Service Map...'.
> Service Map is part of Azure Monitor for VMs, there is no need for redundant onboarding message.

The onboarding message appears if there isn't a `VMConnection` table, if you can suggest a better onboarding message then ping me a better one. Thanks.

File: Workbooks/categoryResources.json [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
-            "order": 100
-          }
-        }      
-      }
-    ]
-  }
\ No newline at end of file
+	"categories": [
+		{
+			"key": "YourUniqueCategoryKey",
```
[the remark this answers is not in the document]
Comment (andrewki-msft) [reply Workbooks/categoryResources.json]: Yes, I wanted to preserve the example category as referenced in the `README.md`

**Comment by nasundar** [comment]
The logic of parsing the "ResourceId" passed in context and picking the subscription fails, if the user does not have any workspaces in the "parsed subscription". Please make this foolproof by allowing the user to pick any subscription he wants.

image

**Comment by nasundar** [comment]
Do some analysis on what is behavior for users having 1000's of subscriptions?  will they be hit with a huge loading time?

**Comment by andrewki-msft** [comment]
> Do some analysis on what is behavior for users having 1000's of subscriptions? will they be hit with a huge loading time?

Changed this to only show default subscriptions. I thought it would be good to allow user to have more choice but they can always change their default subscriptions through the side panel.

**Comment by andrewki-msft** [comment]
> The logic of parsing the "ResourceId" passed in context and picking the subscription fails, if the user does not have any workspaces in the "parsed subscription". Please make this foolproof by allowing the user to pick any subscription he wants.

@nasundar Per our discussion, you were testing in a different environment which led to this issue. This workbook should only be visible under Azure Monitor and AtScale VM Insights. Thanks.

**Comment by andrewki-msft** [comment]
> Please change the gallery group of the template per comment.

The duplicate workbooks issue is fixed both in PPE and PROD

image

File: Workbooks/categoryResources.json [hunk 2]
```diff
@@ -1,13 +1,24 @@
...
+					"description": "Category description",
+					"order": 100
+				}
+			}
+		},
+		{
+			"key": "azureMonitorVirtualMachines",
+			"settings": {
+				"en-us": {
+					"name": "Virtual Machines in Azure Monitor",
```
[the remark this answers is not in the document]
Comment (acearun) [reply Workbooks/categoryResources.json]: Can you address this please?

**Comment by acearun** [comment]
@andrewki-msft, please rename the category to 'Virtual Machines' - take out the in Azure Monitor, its implicit from the context.

File: Workbooks/categoryResources.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (andrewki-msft) [reply Workbooks/categoryResources.json]: Yes, will do

**Comment by andrewki-msft** [comment]
@acearun Acknowledged and done, I've updated the category name

image

**Pull request merged**